src/migrateDashboard.ts

```diff
@@ -56,7 +59,22 @@ export function migrateDashboard(
           dashboardLeafKey,
         ];
       } else {
-        content[dashboardLeafKey] = migrateWidget(widget.bookmark, servers);
+        const [migratedWidget, widgetMigrationError] = migrateWidget(
```

Comment:
It would seem more conventional to have `migrateWidget` throw its error instead of returning `[whatItIsSupposedToReturn, anError?]`.

The code here would become:

```
try {
  const migratedWidget = migrateWidget()
} catch (widgetMigrationError) {
  ...
}
```

Reply:
This function always return the body of the migrated widget, even if an error (or should we say warning) occurred. So throwing is not going to be of any help, unless we include the widget body in the thrown Error.